### PR TITLE
Site doc generator: respect Javadoc on type for smallrye-config

### PR DESCRIPTION
**Pull request by snazy**
Considers the Javadoc of the "most relevant" config-mapping type and adds it as markdown to the generated docs.

### CODE REVIEW

File: servers/quarkus-common/src/main/java/org/projectnessie/quarkus/config/QuarkusBigTableConfig.java [hunk 1]
```diff
@@ -21,6 +21,11 @@
 import java.util.Optional;
 import org.projectnessie.versioned.storage.bigtable.BigTableClientsConfig;
 
+/**
+ * When setting {@code nessie.version.store.type=BIGTABLE} which enables Google BigTable as the
+ * version store used by the Nessie server, the following configurations are applicable in
```
Comment (adutra) [inline servers/quarkus-common/src/main/java/org/projectnessie/quarkus/config/QuarkusBigTableConfig.java:26]: Nit: I'd remove the `in combination with {@code nessie.version.store.type}` everywhere as it is a bit redundant.

File: tools/doc-generator/doclet/src/main/java/org/projectnessie/nessie/docgen/MarkdownFormatter.java [hunk 2]
```diff
@@ -0,0 +1,622 @@
...
+    this.element = requireNonNull(element);
+
+    if (commentTree != null) {
+
+      // @see, @deprecated, etc
+      for (DocTree doc : commentTree.getBlockTags()) {
+        switch (doc.getKind()) {
+          case SEE:
+            {
+              // @code
```
Comment (adutra) [inline tools/doc-generator/doclet/src/main/java/org/projectnessie/nessie/docgen/MarkdownFormatter.java:69]: Comment seems off?

Comment (snazy) [reply tools/doc-generator/doclet/src/main/java/org/projectnessie/nessie/docgen/MarkdownFormatter.java:69]: Copied the off-comment :facepalm:

File: tools/doc-generator/doclet/src/test/java/org/projectnessie/nessie/docgen/TestDocGenTool.java [hunk 3]
```diff
@@ -75,7 +75,19 @@ public void docGenTool(@TempDir Path dir) throws Exception {
         .isRegularFile()
         .content()
         .isEqualTo(
-            "| Property | Default Value | Type | Description |\n"
+            "The docs for `my.prefix`. \n"
+                + "\n"
+                + " * Some    \n"
```
[the remark this answers is not in the document]
Comment (snazy) [reply tools/doc-generator/doclet/src/test/java/org/projectnessie/nessie/docgen/TestDocGenTool.java:80]: A `String.trim()` in the code

File: servers/quarkus-common/src/main/java/org/projectnessie/quarkus/config/QuarkusBigTableConfig.java (hunk 1 above)
Comment (snazy) [reply servers/quarkus-common/src/main/java/org/projectnessie/quarkus/config/QuarkusBigTableConfig.java:26]: Yea - removed

**Review by adutra — APPROVED** [review]

**Pull request merged**